R/Analyis_genome_abundances.Rmd

```diff
@@ -0,0 +1,215 @@
+kable(topleft(Counts, c = 10))
+```
+
+### Mapping rate
+
+```{r mapping_rate}
+mapping_rate <- read_tsv("Example/Results/mapping_rate.tsv", 
+                         col_names = c("Sample", "Mapping_rate"), skip = 1)  %>%
+  mutate(tmp = "samples")
+
+ggplot(mapping_rate, aes(x = tmp, y = Mapping_rate)) +
+  geom_jitter() +
+  ylim(c(0, 1))+
+  theme_minimal()
+```
+
+### Stats based on raw counts
+
+There are good reasons to use rawcounts and use centric log ratios, see more in 
+
+Gloor, Gregory B., Jean M. Macklaim, Vera Pawlowsky-Glahn, and Juan J. Egozcue. 2017. “Microbiome Datasets Are Compositional: And This Is Not Optional.” Frontiers in Microbiology 8 (November). Frontiers: 2224. doi:10.3389/fmicb.2017.02224.
+
+
+For differencial abundance analysis see also the same paper.
+
+```{r clr_transform}
+# transform counts with centered log ratio
+data <- Counts %>%
+  column_to_rownames(var = "Sample") %>%
+  as.data.frame()
+data <- transform(data, "clr")
+```
+
+```{r heatmap, fig.height = 7}
+pheatmap(data, cluster_rows = T, cluster_cols = T,
+         labels_row = Tax$Labels)
+```
+
+### PCA (PCoA) of the robust aitchison distance
+
+```{r pcoa_aitchison}
+pcoa_data <- pcoa(dist(t(data), method = "euclidean"))
+pcoa_df <- as.data.frame(pcoa_data$vectors)
+
+plt <- ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = rownames(pcoa_df))) +
+  geom_point() +
+  labs(x = paste0("PC1 (", round(pcoa_data$values$Relative_eig[1] * 100, digits = 1), "%)"),
+       y = paste0("PC2 (", round(pcoa_data$values$Relative_eig[2] * 100, digits = 1), "%)")) +
+  theme_minimal()
+
+ggplotly(plt)
+```
+
+### Relative abundance
+
+For the relative abundance we take the coverage over the genome not the raw counts. This inmplicit normalizes for genome size. The coverage is calculated as the median of the coverage values calculated in 1kb blocks.
+
+```{r}
+D <- read_tsv("Example/Results/counts/median_coverage_genomes.tsv") %>%
+  column_to_rownames(var = "X1") %>%
+  as.data.frame()
+kable(head(D))
+```
+
+```{r relative_abund}
+# calculate relative abundance
+rel_ab <- sweep(D, 1, rowSums(D),`/`)
+```
+
+```{r calc_most_abund}
+# get most abundant genomes
+counts_per_genome <- data.frame(sums = colSums(rel_ab)) %>%
+  rownames_to_column(var = "Sample") %>%
+  left_join(Tax, by = c("Sample" = "user_genome")) %>%
+  arrange(desc(sums))
+  
+
+ggplot(counts_per_genome %>%
+         top_n(sums, n = 10), aes(x = reorder(Labels, sums), y = sums)) +
+  geom_col() +
+  labs(x = "", y = "Abundance [rel_ab]", title = "Most abundant genomes") +
+  theme_minimal() +
+  theme(axis.text.x = element_text(angle = 90))
+```
+
+### Typical bar chart
+
+```{r barchart}
+level <- 'family'
+
+grouped_data <- rel_ab %>%
+  t() %>%
+  as.data.frame() %>%
+  rownames_to_column(var = "user_genome") %>%
+  pivot_longer(cols = -user_genome, names_to = "Sample", values_to = "rel_ab") %>%
+  left_join(Tax, by = "user_genome") %>%
```

Comment:
The above would need to be `scale_fill_manual(...)` instead of `scale_colour_manual()`

You can also invoke brewer palettes using `scale_fill_brewer(palette = "Paired")`, however, they do not recycle.

I added your incantation but commented it out, along with `library(RColorBrewer)` in the library code block. 

I also added a 21 color palette in case you like that, which is knitted in the push I just made to this branch. It's originally specified here: https://www.r-bloggers.com/2013/02/the-paul-tol-21-color-salute/